mm2src/mm2_main/src/lp_wallet.rs

```diff
@@ -220,12 +230,16 @@ async fn decrypt_validate_or_save_passphrase(
             // If an existing passphrase is found and it matches the decrypted passphrase, return it
             Ok(Some(decrypted_passphrase))
         },
-        None => {
+        None if ctx.allow_registrations() => {
             save_encrypted_passphrase(ctx, wallet_name, &encrypted_passphrase_data)
                 .await
                 .mm_err(|e| WalletInitError::WalletsStorageError(e.to_string()))?;
             Ok(Some(decrypted_passphrase))
         },
+        None => {
+            // If no passphrase is found and registrations are not allowed, return an error
+            Err(WalletInitError::WalletCreationNotAllowed.into())
+        },
```

Comment:
This is straightforward and easy to follow for someone reading the code. Same for other places too

```suggestion
        None => {
            if ctx.allow_registrations() { 
              save_encrypted_passphrase(ctx, wallet_name, &encrypted_passphrase_data)
               .await
               .mm_err(|e| WalletInitError::WalletsStorageError(e.to_string()))?;
            return Ok(Some(decrypted_passphrase));
         }
            // If no passphrase is found and registrations are not allowed, return an error
            Err(WalletInitError::WalletCreationNotAllowed.into())
        },
```